app/actions/local/user.ts

```diff
@@ -105,25 +105,25 @@ export const updateRecentCustomStatuses = async (serverUrl: string, customStatus
     });
 };
 
-export const updateUserPresence = async (serverUrl: string, userStatus: UserStatus) => {
-    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
-    if (!operator) {
+export const updateLocalUser = async (serverUrl: string, userId: string, userDetails: Partial<UserModel>) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
         return {error: `${serverUrl} database not found`};
     }
 
-    const user = await queryUserById(operator.database, userStatus.user_id);
-    if (user) {
-        user.prepareUpdate((record) => {
-            record.status = userStatus.status;
+    try {
+        const user = await database.get(MM_TABLES.SERVER.USER).find(userId) as UserModel;
+        await database.write(async () => {
+            await user.update((userRecord: UserModel) => {
+                userRecord.status = userDetails?.status ?? user.status;
+                userRecord.lastPictureUpdate = userDetails?.lastPictureUpdate ?? user.lastPictureUpdate;
```

Comment:
so you replaced the `updateUserPresence` function for something more generic which is fine, but here you are only only updating two values instead of every possible property. it seems to me that this needs more work or you keep them separately.

Also the problem you'll face here is that the argument is a `Partial<UserModel>` basically forcing you to initialize a UserModel in other places as you cannot set it's values within an existing object directly. So in my opinion to avoid that from happening you'll need to pass `Partial<UserProfile>` instead, that way you can quickly create an UserProfile object anywhere and easily.

Reply:
Thanks Elias.  I wanted to have something similar to how the Operator's transformer were mapping the field.  I have updated this code now for it to include the other fields as well.